include/util/meta_string.hpp

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright (c) 2022, Alibaba Group Holding Limited;
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Author's Email: metabeyond@outlook.com
+ * Author's Github: https://github.com/refvalue/
+ * Description: this source file contains an implementation of a compile-time
+ * meta-string that can be utilized as a non-type parameter of a template.
+ */
+#pragma once
+
+#include <algorithm>
+#include <array>
+#include <compare>
+#include <concepts>
+#include <cstddef>
+#include <span>
+#include <string_view>
+#include <utility>
+
+namespace refvalue {
+template <std::size_t N>
+struct meta_string {
```

Comment:
lack of concat function and operator +

Reply:
a data() member function is also needed.

Reply:
> **PRETTY_FUNCTION** does not contain template function arguments symbols, only contain function name, so i think we can't support template rpc function now, and i think it is acceptable.
> 
> So remove template function test case is fine.

OK. I have implemented a new version of uniform output without template arguments.